tests/unit/test_io_utils.py

```diff
@@ -0,0 +1,218 @@
+        testing_out = (
+            """{\n    "id": "4bf7ca9c-0b39-48fb-9af7-83e3e488f62b",\n"""
+            """    "status": "Submitted"\n}\n"""
+        )
+
+        # Here the function is being run and allows us to redirect the stdout which
+        # would be what the function prints to the screen to file like object
+        func_stdout = io.StringIO()
+        with redirect_stdout(func_stdout):
+            io_utils.pretty_print_json(
+                """{"id": "4bf7ca9c-0b39-48fb-9af7-83e3e488f62b",
+                "status": "Submitted"}"""
+            )
+
+        # assert the function stdout is the same as the expected out
+        assert func_stdout.getvalue() == testing_out
+
+    def test_create_directory(self, temp_dir_path):
+
+        test_io_utility_temp_folder = Path(temp_dir_path + "/test_io_utility/")
+
+        # Delete test dir if exists from previous test run
+        if test_io_utility_temp_folder.exists():
+            shutil.rmtree(test_io_utility_temp_folder)
+
+        # Test that function is able to create a folder.
+        io_utils.create_directory(
+            dir_path=test_io_utility_temp_folder, parents=False, exist_ok=False
+        )
+        assert (
+            test_io_utility_temp_folder.exists()
+        ), "Temp folder should have been created"
+
+        # Delete temp folder
+        shutil.rmtree(test_io_utility_temp_folder)
+
+    def test_create_directory_exist_ok(self, temp_dir_path):
+        test_io_utility_temp_folder = Path(temp_dir_path + "/test_io_utility/")
+
+        # Delete test dir if exists from previous test run
+        if test_io_utility_temp_folder.exists():
+            shutil.rmtree(test_io_utility_temp_folder)
+
+        # Create test_io_utility_temp_folder
+        os.mkdir(test_io_utility_temp_folder)
+
+        # Test that function fails and raises an error because `exist_ok`
+        # is set to `False` and the folder its being asked to create already
+        # exists from the previous lines.
+        with pytest.raises(FileExistsError):
+            io_utils.create_directory(
+                dir_path=test_io_utility_temp_folder, parents=False, exist_ok=False
+            ), "Should fail because folder already exists from previous test"
+
+        # Test that exception is not raised in this case even though
+        # folder already exists, because `exist_ok` variable is set to `True`.
+        io_utils.create_directory(
+            dir_path=test_io_utility_temp_folder, parents=False, exist_ok=True
+        )
+
+        # Delete temp folder
+        shutil.rmtree(test_io_utility_temp_folder)
+
+    def test_create_directory_parents(self, temp_dir_path):
+
+        test_io_utility_temp_folder = Path(temp_dir_path + "/test_io_utility/")
+        test_io_utility_temp_folder_parents = Path(
+            temp_dir_path + "/test_io_utility/parent1/parent2/"
+        )
+
+        # Delete test dir if exists from previous test run
```

Comment:
This should be unecessary if getting the temp file is correct since it should always be a new temp directory on each run.  If that's not the case we should fix that instead of this kludge.

Reply:
Made changes the all the functions here to use "[tmp_path](https://docs.pytest.org/en/latest/how-to/tmpdir.html)" which is a pytest fixture that provides a temp dir per function